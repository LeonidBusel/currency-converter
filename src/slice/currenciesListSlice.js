import { createSlice } from '@reduxjs/toolkit'

export const currenciesListSlice = createSlice({
    name: 'currenciesList',
    initialState: {
        isFetching: false,
        isLoad: false,
        list: []
    },
    reducers: {
        currenciesListFetch: state => {
            state.isFetching = true;
        },
        currenciesListFetchSuccess: (state, { payload }) => {
            state.isFetching = false;
            state.isLoad = true;
            state.list = payload.slice();
        },
        currenciesListFetchFail: (state) => {
            state.isFetching = false;
            state.list = [];
        }
    }
})

export const { currenciesListFetch, currenciesListFetchSuccess, currenciesListFetchFail } = currenciesListSlice.actions

export default currenciesListSlice.reducer