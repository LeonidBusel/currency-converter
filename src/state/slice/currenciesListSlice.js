import { createSlice } from '@reduxjs/toolkit'

export const initialState = {
    isFetching: false,
    isLoad: false,
    list: []
};

export const currenciesListSlice = createSlice({
    name: 'currenciesList',
    initialState,
    reducers: {
        currenciesListFetch: state => {
            state.isFetching = true;
        },
        currenciesListFetchSuccess: (state, { payload }) => {
            state.isFetching = false;
            state.isLoad = true;
            state.list = payload.slice();
        },
        currenciesListFetchFail: (state) => {
            state.isFetching = false;
            state.list = [];
        }
    }
})

export const { currenciesListFetch, currenciesListFetchSuccess, currenciesListFetchFail } = currenciesListSlice.actions

export default currenciesListSlice.reducer