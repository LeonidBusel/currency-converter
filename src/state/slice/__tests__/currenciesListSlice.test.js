import currenciesListSlice, { initialState, currenciesListFetch, currenciesListFetchSuccess, currenciesListFetchFail } from "../currenciesListSlice";

const modifyState = {
    "isFetching": false,
    "isLoad": true,
    "list": [1, 2]
}

describe('currenciesListSlice', () => {
    it('handle currenciesListFetch correctly', () => {
        const actual = currenciesListSlice(initialState, currenciesListFetch());

        expect(actual).toMatchSnapshot();
    });

    it('handle currenciesListFetchSuccess correctly', () => {
        const actual = currenciesListSlice(initialState, currenciesListFetchSuccess([1, 2]));

        expect(actual).toMatchSnapshot();
    });

    it('handle currenciesListFetchFail correctly', () => {
        const actual = currenciesListSlice(modifyState, currenciesListFetchFail());

        expect(actual).toMatchSnapshot();
    });
});