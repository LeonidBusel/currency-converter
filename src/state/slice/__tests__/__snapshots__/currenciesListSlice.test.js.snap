// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currenciesListSlice handle currenciesListFetch correctly 1`] = `
Object {
  "isFetching": true,
  "isLoad": false,
  "list": Array [],
}
`;

exports[`currenciesListSlice handle currenciesListFetchFail correctly 1`] = `
Object {
  "isFetching": false,
  "isLoad": true,
  "list": Array [],
}
`;

exports[`currenciesListSlice handle currenciesListFetchSuccess correctly 1`] = `
Object {
  "isFetching": false,
  "isLoad": true,
  "list": Array [
    1,
    2,
  ],
}
`;
